### ISSUE TITLE
Should we have a Model State Token?

### ISSUE THREAD

**Issue by markusvoelter** [LionWeb-io/specification#68]
We are currently implementing the Lionweb Bulk API for modelix. Since modelix is multi-user, the contents of the repository can have changed by the time the client sends an update request. If we don't do anything about this, the data will be overwritten. 

There is a whole set of different approaches around versioning and locking, and we should discuss these soon in the context of Lionweb.

In the meantime, we implement the following approach in our code: 

* When the client reads data from the server, the server also sends an opaque
  token to the client.
* When the client sends an update, then
  - if the client does not send back the token, the server refuses the update
  - if the client sends back the token, the server (somehow) uses that token
    to find out of if the data has changed in the meantime. If no, then the 
    update is successful. If it has changed, then the server refuses the update.
* if the server sends no such token, then the prevent-overwriting-data feature
  is not supported.

This approach avoids the need for the server to maintain a session for each client. It also maps very nicely to modelix -- it maintains such a global version anyway. But it allows any other tool to put any other information -- arbitrary Json! --  into that token.

Long story short: we have to discuss this further :-)

# Decision 2024-10-04: Not needed

We don't enforce perfectly synced model state, as the repo can solely decide on merge strategy.

**Comment by enikao** [comment]
Pro:
* Check whether my client model is up-to-date with repo
* Could enforce "pull-before-push" semantics

Con:
* Might be additional effort to maintain for repo
* Not strictly required
